### ISSUE TITLE
Schema params and attribute types

### ISSUE THREAD

**Issue by harlan-zw** [GoogleChromeLabs/third-party-capital#21]
Nuxt is very heavily invested in TypeScript. Currently, the exported schemas do not have data that could be used to generate any meaningful types for the params or attributes.

Ideally, the schemas would expose these types.

As a nice to have, it would be cool to implement [Valibot](https://www.builder.io/blog/introducing-valibot) or alternative.

**Comment by harlan-zw** [comment]
This issue can be considered resolved when the types validate the user input to initiate the script.

```ts
export const GoogleAnalytics = ({ ...args }: Inputs) => {
  return formatData(data as Data, args);
};
```

```ts
// nothing is validated
GoogleAnalytics({
  foo: 'bar',
  id: null,
})
```

**Comment by huang-julien** [comment]
Yes, I wasn't thinking about something too deep with valibot/zod/yup. Would be happy to open few PRs providing types if we migrate JSON to TS. For example something like:

```ts
interface GoogleAnalyticsQueries {
  id: string
}
```

**Comment by flashdesignory** [comment]
@huang-julien - do you mean something like the options we currently have exported?
https://github.com/GoogleChromeLabs/third-party-capital/blob/main/src/types/type-declarations.ts#L63
https://github.com/GoogleChromeLabs/third-party-capital/blob/main/src/types/type-declarations.ts#L84

we can certainly expand on those

**Comment by huang-julien** [comment]
oh, indeed :eyes:
